pkg/util/hostname.go

```diff
@@ -142,13 +150,13 @@ func GetHostnameData(ctx context.Context) (HostnameData, error) {
 	var err error
 	var provider string
 
-	// try the name provided in the configuration file
+	// Try the name provided in the configuration file
 	configName := config.Datadog.GetString("hostname")
 	err = validate.ValidHostname(configName)
 	if err == nil {
 		hostnameData := saveHostnameData(cacheHostnameKey, configName, HostnameProviderConfiguration)
 		if !isHostnameCanonicalForIntake(ctx, configName) && !config.Datadog.GetBool("hostname_force_config_as_canonical") {
-			_ = log.Warnf("Hostname '%s' defined in configuration will not be used as the in-app hostname. For more information: https://dtdg.co/agent-hostname-force-config-as-canonical", configName)
+			_ = log.Warnf(CanonicalHostnameWarning, configName)
```

Comment:
Would it make sense to extract this whole block (from `saveHostnameData`) into a function instead? (and keep format string in the `Warnf` call).

Reply:
Yeah I see what you mean. Let me take a look at this code again later today and see if this can be done without making things harder to maintain. When I was originally working on the PR, it seemed like the extraction would un-necessarily make logic even more convoluted in this file but I'll give your suggestion a try.

Reply:
Potential diff on this: https://gist.github.com/sgnn7/c1440b8531081c1ba0975bdfb4a19f82. Let me know what you think.

Reply:
Updated with the change requested